utils/env_preferences/src/apple.rs

```diff
@@ -0,0 +1,132 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+pub mod apple_prefs {
+    use core_foundation_sys::{
+        array::{CFArrayGetCount, CFArrayGetValueAtIndex},
+        base::{CFRelease, CFRetain},
+        calendar::{CFCalendarCopyCurrent, CFCalendarCopyLocale, CFCalendarGetIdentifier},
+        locale::{CFLocaleCopyPreferredLanguages, CFLocaleGetIdentifier},
+        string::CFStringGetCStringPtr,
+    };
+    use std::ffi::CStr;
+
+    pub fn get_locales() -> Vec<String> {
+        let mut languages: Vec<String> = Vec::new();
+
+        // SAFETY: The call to `CFLocaleCopyPreferredLanguages` returns an immutable reference to `CFArray` which is owned by us
```

Comment:
if we own the `CFArray`, do we need to free it later?

Reply:
Sure thing